src/transformStateWithClones.js

```diff
@@ -7,7 +7,47 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+
+  const resultArray = [];
+
+  for (const action of actions) {
+    let newState = { ...currentState };
+
+    switch (action.type) {
```

Comment:
Use a destruction for `action`
```suggestion
  for (const action of actions) {
    let newState = { ...currentState };
    const { type, .......... } = action;

    switch (type) {
```